pandas/core/reshape/merge.py

```diff
@@ -1479,10 +1474,10 @@ def restore_dropped_levels_multijoin(
     left: MultiIndex,
     right: MultiIndex,
     dropped_level_names,
-    join_index,
-    lindexer,
-    rindexer,
-):
+    join_index: MultiIndex,
+    lindexer: np.ndarray,
+    rindexer: np.ndarray,
+) -> Tuple[List[Index], np.ndarray, List[str]]:
```

Comment:
Can you update the docstring to match for the return type of 'names of combined multiindexes'

Reply:
Updated docstring. Plan to take a much deeper dive into this code in followups, will try to clean up more inconsistencies once I have a better handle on what's going on here.